tests/test_mllam_dataset.py

```diff
@@ -0,0 +1,131 @@
+# Standard library
+import os
+
+# Third-party
+import pooch
+
+# First-party
+from create_mesh import main as create_mesh
+from neural_lam.config import Config
+from neural_lam.utils import load_static_data
+from neural_lam.weather_dataset import WeatherDataset
+from train_model import main as train_model
+
+os.environ["WANDB_DISABLED"] = "true"
+
+
+def test_retrieve_data_ewc():
+    # Initializing variables for the client
+    S3_BUCKET_NAME = "mllam-testdata"
+    S3_ENDPOINT_URL = "https://object-store.os-api.cci1.ecmwf.int"
+    S3_FILE_PATH = "neural-lam/npy/meps_example_reduced.v0.1.0.zip"
+    S3_FULL_PATH = "/".join([S3_ENDPOINT_URL, S3_BUCKET_NAME, S3_FILE_PATH])
+    known_hash = (
+        "98c7a2f442922de40c6891fe3e5d190346889d6e0e97550170a82a7ce58a72b7"
+    )
+
+    pooch.retrieve(
+        url=S3_FULL_PATH,
+        known_hash=known_hash,
+        processor=pooch.Unzip(extract_dir=""),
+        path="data",
+        fname="meps_example_reduced.zip",
+    )
+
+
+def test_load_reduced_meps_dataset():
+    data_config_file = "data/meps_example_reduced/data_config.yaml"
+    dataset_name = "meps_example_reduced"
+
+    dataset = WeatherDataset(dataset_name="meps_example_reduced")
+    config = Config.from_file(data_config_file)
+
+    var_names = config.values["dataset"]["var_names"]
+    var_units = config.values["dataset"]["var_units"]
+    var_longnames = config.values["dataset"]["var_longnames"]
+
+    assert len(var_names) == len(var_longnames)
+    assert len(var_names) == len(var_units)
+
+    # TODO: can these two variables be loaded from elsewhere?
+    n_grid_static_features = 4
```

Comment:
I think you can assume here the number of "grid static features" that is present in the `grid_static.pt` file https://github.com/mllam/neural-lam/blob/5b71be3c68d815e0e376ee651c14f09d801f86de/neural_lam/utils.py#L56

The `WeatherDataset` doesn't currently expose number of grid-features, static or dynamic. Maybe that is something we should consider in future @sadamov?

Reply:
or maybe these config file should at least explicitly state which grid static features there are?

Reply:
We can define paths to zarr archives from static, forcing, boundary, state directly in the future (see  https://github.com/mllam/neural-lam/blob/feature_dataset_yaml/neural_lam/data_config.yaml)

`num_data_vars` is already implemented in the Config class: https://github.com/mllam/neural-lam/blob/5b71be3c68d815e0e376ee651c14f09d801f86de/neural_lam/config.py#L51

Reply:
ok, in that case I think this comment in the code can stay, or you could maybe change it to say `in future the number of grid static features will be provided by the Dataset class itself` or something like that. Sound ok?